main.go

```diff
@@ -111,29 +115,40 @@ func cloneRepo(cfg repoConfig) error {
 		HTTPUsername: string(cfg.GitHTTPUsername),
 		HTTPPassword: string(cfg.GitHTTPPassword),
 	}); err != nil {
-		return err
+		return newStepError(
+			"activate_git_http_credentials_failed",
+			err,
+			fmt.Sprintf("Activating Git HTTP credentials for %s failed", cfg.RepositoryURL),
+		)
 	}
 
 	// Git clone
 	logger := logv2.NewLogger()
 	envRepo := env.NewRepository()
 
-	tracker := tracker.NewStepTracker(envRepo, logger)
+	stepTracker := tracker.NewStepTracker(envRepo, logger)
 	cmdFactory := cmdv2.NewFactory(envRepo)
 	// patchSource and mergeRefChecker used for merging only
-	// build URL and build api token doesn't apply here
+	// build URL and build api token don't apply here
 	patchSource := bitriseapi.NewPatchSource("", "")
-	mergeRefChecker := bitriseapi.NewMergeRefChecker("", "", retry.NewHTTPClient(), logger, tracker)
-	gitcloner := gitclone.NewGitCloner(logger, tracker, cmdFactory, patchSource, mergeRefChecker)
+	mergeRefChecker := bitriseapi.NewMergeRefChecker("", "", retry.NewHTTPClient(), logger, stepTracker)
+	gitcloner := gitclone.NewGitCloner(logger, stepTracker, cmdFactory, patchSource, mergeRefChecker)
 	config := gitclone.Config{
 		RepositoryURL: cfg.RepositoryURL,
-		CloneIntoDir:  cfg.CloneIntoDir, // Using same directory later to run scan
+		CloneIntoDir:  cfg.CloneIntoDir, // Using the same directory later to run scan
 		Branch:        cfg.Branch,
 
 		UpdateSubmodules: true,
 	}
 	if _, err := gitcloner.CheckoutState(config); err != nil {
-		return err
+		hasSSH := len(cfg.SSHRsaPrivateKey) > 0
```

Comment:
It used to be that git-clone returned this newStepError, for recommending available branches (https://github.com/bitrise-steplib/steps-git-clone/blob/b37dd5bbbfb17bd29d1e17eeb231f352b0d57824/gitclone/git.go#L137) and just for analytics as well. Does this mean that is no longer needed, and we can remove newStepError from the git-clone Step (not a pretty solution currently so I do not mind). It is possible that the branch recommendation is also not needed as the web UI gives the option to select branches now.

Reply:
I am not fully up-to-date with the history of the `newStepError` and actually did not check if the git step already returns this. So I am not sure I can answer the `we can remove newStepError from the git-clone Step` question. My feeling is yes because the add new app flow has changed significantly. But the reason I introduced the extra `newStepError`s because the git clone step has a path where it returns only regular errors. Errors such as these:
```
- failed to fetch branch (refs/heads/master): remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/gaborszakacs/Fruta.git/': please make sure the branch still exists

- failed to fetch branch (refs/heads/main): fatal: couldn't find remote ref refs/heads/main: please make sure the branch still exists

- failed to fetch branch (refs/heads/dev-automation): git@bitbucket.org: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.: please make sure the branch still exists
```
But now that you showed me that the git step also returns these errors I will first check the returned error and if it is a `newStepError` type then I will not do anything. And in the other cases convert it to a `newStepError`.

About the branch recommendation, I did not know that the git step does this. It is probably not needed anymore as the UI offers the same feature for the users.